src/TimeoutMigrationTool/Asp/AspTimeoutsSource.cs

```diff
@@ -0,0 +1,608 @@
+
+            var toolState = await TryLoadToolStateEntity(tableClient);
+
+            var toolStateTable = await GetAndCreateToolStateTableIfNotExists(tableClient);
+
+            var endpointTimeoutTable = await GetEndpointTimeoutTable(toolState.EndpointName, tableClient);
+
+            var lowerLimitForCuttOffToFilterGuidEntriesOut = DateTime.UtcNow.AddYears(-1).ToString(partitionKeyScope);
+            var upperLimitForCuttOffToFilterGuidEntriesOut = DateTime.UtcNow.AddYears(100).ToString(partitionKeyScope);
+
+            var query = new TableQuery<PartialTimeoutDataEntityWithOwningTimeoutManager>()
+                .Where(CombineFilters(
+                    CombineFilters(GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, lowerLimitForCuttOffToFilterGuidEntriesOut),
+                        TableOperators.And,
+                        GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual, upperLimitForCuttOffToFilterGuidEntriesOut)
+                            ),
+                    TableOperators.And,
+                    GenerateFilterCondition("OwningTimeoutManager", QueryComparisons.Equal, $"{AspConstants.MigrationOngoingPrefix}{toolState.EndpointName}")));
+
+            query.SelectColumns = new List<string> { "OwningTimeoutManager" };
+            TableContinuationToken token = null;
+            CancellationToken queryCancellationToken = CancellationToken.None;
+
+            do
+            {
+                // will issue a table scan for now
+                var seg = await endpointTimeoutTable.ExecuteQuerySegmentedAsync(
+                        query: query,
+                        token: token,
+                        requestOptions: null,
+                        operationContext: null,
+                        cancellationToken: queryCancellationToken)
+                    .ConfigureAwait(false);
+                token = seg.ContinuationToken;
+
+                foreach (var t in seg.Results.GroupBy(x => x.PartitionKey))
+                {
+                    // entries with Guid as partition key should never be modified. We are only interested in the query entities
+                    if (Guid.TryParse(t.Key.AsSpan(), out _))
+                    {
+                        continue;
+                    }
+
+                    // with batching we can at least efficiently insert
+                    var batch = new TableBatchOperation();
+                    foreach (var partialTimeoutDataEntity in t)
+                    {
+                        var currentOwningTimeoutManager = partialTimeoutDataEntity.OwningTimeoutManager;
+
+                        partialTimeoutDataEntity.ETag = "*";
+                        partialTimeoutDataEntity.OwningTimeoutManager =
+                            currentOwningTimeoutManager.Replace(AspConstants.MigrationOngoingPrefix, string.Empty);
+
+                        if (batch.Count == 100)
```

Comment:
Check for size?

Reply:
https://github.com/Particular/TimeoutMigrationTool/pull/172#discussion_r570014836